### PR TITLE
Add breadcrumb rich data on the frontpage

### LINKED ISSUE #22416: SEO-Problem: Breadcrumblist error & "Offers", "review" or "aggregateRating" must be specified
#### Describe the bug
**itemListElement** in breadcrumblist is missing on some pages (seems it is fine on the product pages).
With the update to 1.7.7 google is showing a problem with missing **itemListElement** and some other bugs like:

`"Offers", "review" or "aggregateRating" must be specified`

 on some pages, it seems basically on the startpage and also some other category pages

#### Expected Behaviour 
would be to have full breadcrumblist with **itemListElement** and **Offers**, **review** or **aggregateRating** had to be specified on all pages available

#### Steps to reproduce
You would need to check the code, please also check my screenshot from the startpage of my prestashop installation, which is showing the "missing **itemListElement**" bug and displaying "**Offers**, **review** or **aggregateRating** must be specified" bug

1. Open this site https://search.google.com/test/rich-results
2. Paste your FO link
3. See errors

#### Screenshots
**"Offers", "review" or "aggregateRating" must be specified** bug:

https://drive.google.com/file/d/16aBG8tfPRGuo6XpHqh-C7k3E62p6sLLY/view

**missing "itemListElement"** bug:

https://drive.google.com/file/d/1kerCmPxQkaJ-wuP6Hs-pMnLFTwVfYH9T/view

ps177_breadcrumbserror

#### Additional infos

- PrestaShop version: 1.7.7
- PHP version: 7.3
- Multilanguage
- Multishop

--> there is no technical problem - this is a SEO topic!
--> it seems also to be connected to multishops - as i did check it with another shop which is also mulitlanguage with the same languages, just not multishop installation.

### PR DESCRIPTION
**Pull request by tswfi**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add rich data to special case of frontpage in breadcrumbs
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #22416
| How to test?  | Use google rich data test tool: https://search.google.com/test/rich-results

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22417)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: themes/classic/templates/_partials/breadcrumb.tpl [hunk 1]
```diff
@@ -33,8 +33,9 @@
               <meta itemprop="position" content="{$smarty.foreach.breadcrumb.iteration}">
             </li>
           {elseif isset($path.title)}
-            <li>
-              <span>{$path.title}</span>
+            <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
```
Comment (PululuK) [inline themes/classic/templates/_partials/breadcrumb.tpl]:
```suggestion
            <li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
```

**Review by PululuK — DISMISSED** [review]
Maybe we should pass all the `itemtype` to `https`

File: themes/classic/templates/_partials/breadcrumb.tpl (hunk 1 above)
Comment (tswfi) [reply themes/classic/templates/_partials/breadcrumb.tpl]: True, this is probably originally from time when schema.org always used http: https://schema.org/docs/faq.html#19

Looks like there are some inconsistencies on the usage around the templates already. 

Maybe another issue and pr for making them all consistent?

Comment (tswfi) [reply themes/classic/templates/_partials/breadcrumb.tpl]: https://github.com/PrestaShop/PrestaShop/search?q=itemtype%3D%22

7 places using https 
9 using http

Comment (PululuK) [reply themes/classic/templates/_partials/breadcrumb.tpl]: > Looks like there are some inconsistencies on the usage around the templates already.

Indeed, you are right !

> Maybe another issue and pr for making them all consistent?

Yes you can provide another PR to fix it.

Thanks you @tswfi

**Review by PululuK — DISMISSED** [review]

File: themes/classic/templates/_partials/breadcrumb.tpl (hunk 1 above)
Comment (tswfi) [reply themes/classic/templates/_partials/breadcrumb.tpl]: @PululuK what is the process? should there always be a issue first and then a pr? (this is my second pr to the prestashop project so I'm still trying to understand the process :)

Comment (tswfi) [reply themes/classic/templates/_partials/breadcrumb.tpl]: Created new issue for this: #22418 pr in a moment.

**Review by matthieu-rolland — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: themes/classic/templates/_partials/breadcrumb.tpl (hunk 1 above)
Comment (NeOMakinG) [reply themes/classic/templates/_partials/breadcrumb.tpl]: Yeah, every PR needs an issue, thanks for your contribs @tswfi :)

**Review by NeOMakinG — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi!

This PR cannot be tested.
The issue should be validated first.
Ping @PrestaShop/product-team 

Thanks!

**Comment by tswfi** [comment]
> This PR cannot be tested.

Should be easily testable:
* set up shop without this pr.
* use google test tool to check shop frontpage: https://search.google.com/test/rich-results
 => should give error
* apply this pr
* retest
 => should not give error

This can also be tested by looking at the frontpage rendered html code to see if itemtype and itemprop attributes are defined for the breadcrumb rich data.

Not sure how to write a real automated test case for this as the checks should really be done with 3rd party tools that check all the rich data (there probably exists a tool to do this but I don't know about it).

**Comment by marionf** [comment]
I remove the waiting for PM label as the issue is a bug and not an improvement
We need to reproduce the issue first
Let me know after if a PM opinion is needed ;)

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM !!
Before PR
image
image


After PR
image
image


Tested on multiple FO page

Thanks @tswfi !!

**Comment by Progi1984** [comment]
Thanks @tswfi & @SD1982

**Comment by alfonsolr09** [comment]
image
new error with using the code of new PR
Error on www. your website /shops

**Comment by tswfi** [comment]
@alfonsolr09 this pr has already been merged, you probably want to open a new issue.

**Comment by alfonsolr09** [comment]
already did

**Pull request merged**